gplog/gplog.go

```diff
@@ -414,6 +409,36 @@ func Fatal(err error, s string, v ...interface{}) {
 	}
 }
 
+/*
+ * The Custom log function allows a caller to set different verbosity thresholds for logging to the shell or logfile
+ */
+
+func Custom(customFileVerbosity int, customShellVerbosity int, s string, v ...interface{}) {
+	logMutex.Lock()
+	defer logMutex.Unlock()
+	var message string
+	if logger.fileVerbosity >= customFileVerbosity {
+		message = GetLogPrefix(getVerbosityString(customFileVerbosity)) + fmt.Sprintf(s, v...)
+		_ = logger.logFile.Output(1, message)
+	}
+	if customShellVerbosity == LOGERROR {
```

Comment:
does this need to be treated separately? I believe `ERROR` has the code of `0` so if only the `else` condition is used `logger.shellVerbosity >= customShellVerbosity`, then a custom message with ERROR level will always be `<=` to whatever shellVerbosity is set to. In other words, an ERROR `customShellVerbosity` will always be logged no matter what `logger.shellVerbosity` is set to since `logger.shellVerbosity >=0` as long as `logger.shellVerbosity` is not negative.

If this is only done for colorization of ERROR, then I'd understand. However, this method cannot be used for WARNINGs, and the colorization of WARN will not happen properly, as there is no concept of WARN level (verbosity), only a helper function `Warn` that does some custom stuff without respect to logging levels and thresholds.